src/Rings/mpoly-graded.jl

```diff
@@ -2440,3 +2440,109 @@ function AbstractAlgebra.promote_rule(::Type{MPolyElem_dec{S, T}}, ::Type{U}) wh
     return Union{}
   end
 end
+
+#create homogenous polynomials in graded rings
+#TODO: make this work for non-standard gradings
+@doc Markdown.doc"""
+    rand(S::MPolyRing_dec, term_range, deg_range, v...)
+
+Create a random homogenous polynomial with a random number of
+terms (`rand(term_range)`) and of random degree (`rand(deg_range)`)
+and random coefficients via `v...`.
+"""
+function rand(S::MPolyRing_dec, term_range, deg_range, v...)
+  f = S.R(0)
+  d = rand(deg_range)
+  for i=1:rand(term_range)
+    td = 0
+    t = S.R(1)
+    for j=1:ngens(S.R)-1
+      t *= gen(S.R, j)^rand(0:d-(is_zero(t) ? 0 : total_degree(t)))
+    end
+    t *= gen(S.R, ngens(S.R))^(d-(is_zero(t) ? 0 : total_degree(t)))
```

Comment:
Why is this not in the loop above? Isn't it the same as what the loop body does?

Reply:
It's not. The final expo is not random